### ISSUE TITLE
Avoid `- null` values on templating.

### ISSUE THREAD

**Issue by stafot** [shyiko/kubetpl#8]
I am not sure if the issue I will point is a minor bug or a feature request. Anyway I am using your `Kind: Template`  way of templating,  and in cases that I have non-required ENVs `kubetpl` adds `- null` value for the non-declared ENVs. This output cannot passed directly to `kubectl` as this output is not a valid input. 
My simple workaround for this case was to add a `sed` pipe (`sed '/- null/d'`) in the middle that deletes all lines that have `- null` value and this way I produce a valid input which can be passed to `kubectl create -f -`
Example to reproduce: 
```
# kubetpl:syntax:template-kind

kind: Template
apiVersion: v1
metadata:
  name: dummy-yiic-template
  annotations:
    description: dummyyiic one off template
objects:
- apiVersion: batch/v1
  kind: Job
  metadata:
    name: dummyoneoff-$(COMMAND)
  spec:
   backoffLimit: 5
   activeDeadlineSeconds: 600
   template:
     metadata:
       name: dummyoneoff-$(COMMAND)
       # These are the labels that pods will be given when created via this Job
       labels:
        app: dummy
        tier: cli
        framework: yii
        version: $(BRANCH_AND_VERSION)
     spec:

       containers:
       - name: yiic
         image: dummy/dummy_fpm
         imagePullPolicy: IfNotPresent
         volumeMounts:
         - name: app-volume
           mountPath: /var/www/dummy
         resources:
           requests:
             cpu: $(CPU_REQUEST)
             memory: $(MEM_REQUEST)
           limits:
             cpu: $(CPU_LIMIT)
             memory: $(MEM_LIMIT)
         envFrom:
         - configMapRef:
             name: dummy
         - secretRef:
             name: dummy
         command: ["./yiic", $(COMMAND), $(ACTION), $(ARGS)]
       restartPolicy: Never
       volumes:
       - name: app-volume
         emptyDir: {}
parameters:
- name: BRANCH_AND_VERSION
  description: Branch and Version
  required: true
  parameterType: string
- name: REPLICAS
  description: Number of replicas
  value: 1
  required: true
  parameterType: int
- name: CPU_REQUEST
  description: CPU Request
  required: true
  value: "200m"
  parameterType: string
- name: CPU_LIMIT
  description: CPU Limit
  required: true
  value: 1
  parameterType: string
- name: MEM_REQUEST
  description: Memory Request
  required: true
  value: "256Mi"
  parameterType: string
- name: MEM_LIMIT
  description: Memory Limit
  required: true
  value: "1Gi"
  parameterType: string
- name: COMMAND
  description: Command
  parameterType: string
- name: ACTION
  description: Action
  parameterType: string
- name: ARGS
  description: Arguments
  parameterType: string
```

Fails:

`kubetpl render dummy-oneoff-yiic-template.yml -s BRANCH_AND_VERSION=test.yolo -s COMMAND=devops -s ACTION=demo | kubectl create -f -`
`error: error validating "STDIN": error validating data: ValidationError(Job.spec.template.spec.containers[0].command): unknown object type "nil" in Job.spec.template.spec.containers[0].command[3]; if you choose to ignore these errors, turn validation off with --validate=false`

Succeed:

` kubetpl render dummy-oneoff-yiic-template.yml -s BRANCH_AND_VERSION=test.yolo -s COMMAND=devops -s ACTION=demo |  sed '/- null/d'  | kubectl create -f -`
`job.batch "dummyoneoff-devops" created`

**Comment by shyiko** [comment]
Fixed in [0.7.0](https://github.com/shyiko/kubetpl/blob/master/CHANGELOG.md#070---2018-07-15) :balloon:

**Comment by stafot** [comment]
@shyiko. Awesome, thanks!
